aten/src/ATen/native/cpu/UpSampleKernelAVXAntialias.h

```diff
@@ -530,25 +665,86 @@ void ImagingResampleHorizontalConvolution8u4x(
 
     // Write the output into single uint32
     // (a b c d) -> x_uint32
-    lineOut0[out_x] = _mm_cvtsi128_si32(_mm256_extracti128_si256(sss0, 0));
-    lineOut1[out_x] = _mm_cvtsi128_si32(_mm256_extracti128_si256(sss0, 1));
-    lineOut2[out_x] = _mm_cvtsi128_si32(_mm256_extracti128_si256(sss1, 0));
-    lineOut3[out_x] = _mm_cvtsi128_si32(_mm256_extracti128_si256(sss1, 1));
+    auto o0 = _mm_cvtsi128_si32(_mm256_castsi256_si128(sss0));
+    auto o1 = _mm_cvtsi128_si32(_mm256_extracti128_si256(sss0, 1));
+    auto o2 = _mm_cvtsi128_si32(_mm256_castsi256_si128(sss1));
+    auto o3 = _mm_cvtsi128_si32(_mm256_extracti128_si256(sss1, 1));
+
+    const auto out_x_strided = stride * out_x;
+
+    if (num_channels == 3 && C10_UNLIKELY(out_x_strided + 4 >= max_out_x_strided)) {
+      // Memcpy 4-bytes is faster than 3-bytes and this is a boundary case when we want to write
+      // 4 bytes (R G B | X) to the output buffer (X1 X2 X3 | R1).
+      // The 4th byte in the register (X) has a garbage value and 4th byte in the output buffer (R1) has a correct
+      // value which was preveiously computed by another line. In other words, it means that we can not overwrite
+      // it by simply writing 4 bytes from the register to the output. We'll do the following:
+      //               v----------|
+      // Output = [... X1 X2 X3 | R1 G1 B1 R2 ...]
+      // First, we store next 4 bytes (R1 G1 B1 R2) in a temporary variable
+      // Second, we write 4 bytes from the register to the output: (X1 X2 X3 | R1) -> (R G B | X)
+      // Output = [... R G B | X G1 B1 R2 ...]
+      // Third, we overwrite next 4 bytes of the output (X G1 B1 R2) with stored values (R1 G1 B1 R2)
```

Comment:
Would it make sense to copy just the byte R2?
```
reinterpret_cast<uint8_t*>(&o1)[stride] = lineOut0[out_x_strided + stride];
std::memcpy(lineOut0 + out_x_strided, &o1, 4);
```

Also, it would be good to factor this out since it's copy-pasted 5 times.

Reply:
Let me check this idea, thanks!

Reply:
@peterbell10 I refactored locally the code with this function:
```c++
static void inline _write_endline_rgb_as_uint32(
    uint8_t* C10_RESTRICT output,
    const uint8_t* C10_RESTRICT output_next,
    uint32_t data
) {
  // data is (R G B X), and output_next = (R1 B1 G1 R1 ...)
  // Here we explicitly set X as R1
  uint8_t* data_ptr = reinterpret_cast<uint8_t*>(&data);
  data_ptr[3] = output_next[0];
  std::memcpy(output, data_ptr, 4);
}
```
Any other concerns and also function name is ok for you ?

Reply:
Your comment uses `R1` twice, and `output_next[0]` can just be `output[3]`.

Reply:
Yeah, it should be `// data is (R G B X), and output_next = (R1 B1 G1 R2 ...)`